deployment/npm/install_api.js

```diff
@@ -20,8 +22,24 @@ module.exports = {
     }
 
     const target = getTarget();
-    const sourcePackagePath = path.dirname(require.resolve("@dprint/" + target + "/package.json"));
-    const sourceExecutablePath = path.join(sourcePackagePath, dprintFileName);
+    let sourceExecutablePath;
+    if (target === "unsupported" || process.env.DPRINT_BUILD_FROM_SOURCE === "1") {
+      console.log("Building dprint binary from source...");
+      const packageVersion = require("./package.json").version;
+      const tmp = fs.mkdtempSync(path.join(os.tmpdir(), "tmp-"));
+      const dprintZip = path.join(tmp, "dprint.zip");
+      const dprintRepo = path.join(tmp, `dprint-${packageVersion}`);
+      await downloadFile(
+        `https://github.com/dprint/dprint/archive/refs/tags/${packageVersion}.zip`,
+        dprintZip,
+      );
+      await unzipFile(dprintZip, tmp);
```

Comment:
Maybe it should run `cargo install --locked --version VERSION_GOES_HERE dprint` instead and only if `DPRINT_BUILD_FROM_SOURCE` is set? That way it doesn't need a dependency on yauzl.

Also, what platform are you running on that dprint doesn't have a build for? Maybe we can try to add a distribution for that instead?

Reply:
I am currently working on Arch Linux RISC-V and I would appreciate if you consider adding linux-riscv64-{gnu,musl} distributions for dprint and its native plugins. Besides, running WASM plugins like TypeScript formatter on riscv64 could produce inconsistent results when I tried to build tsc, which definitely needs further investigation...

As for the change, I lost access to my main work PC recently and will get back to it a few days later. :)